lightning/src/chain/chainmonitor.rs

```diff
@@ -570,27 +571,31 @@ where C::Target: chain::Filter,
 	///
 	/// Note that we persist the given `ChannelMonitor` while holding the `ChainMonitor`
 	/// monitors lock.
-	fn watch_channel(&self, funding_outpoint: OutPoint, monitor: ChannelMonitor<ChannelSigner>) -> Result<(), ChannelMonitorUpdateErr> {
+	fn watch_channel(&self, funding_outpoint: OutPoint, monitor: ChannelMonitor<ChannelSigner>) -> ChannelMonitorUpdateStatus {
 		let mut monitors = self.monitors.write().unwrap();
 		let entry = match monitors.entry(funding_outpoint) {
 			hash_map::Entry::Occupied(_) => {
 				log_error!(self.logger, "Failed to add new channel data: channel monitor for given outpoint is already present");
-				return Err(ChannelMonitorUpdateErr::PermanentFailure)},
+				return ChannelMonitorUpdateStatus::PermanentFailure
```

Comment:
we are combining two kind of errors into one,

1. represents ChannelMonitorPersistStatus
2. other is ChannelMonitorUpdateStatus which is overall and can be due to other reasons as well.
do you think we need to distinguish b/w the two?